src/agent/payload.cc

```diff
@@ -44,6 +44,7 @@ using std::string;
 // do instrumentation, the purpose of this is to stop
 // instrumentation when we hit exit function
 std::atomic<int> IN_INSTRUMENTATION {1};
+int trampGuard = 0;
```

Comment:
This should use RAII.  Change `int trampGuard = 0` to `bool inTramp = false`.  Then create a class to that take a bool reference in the constructor that sets it to true (and asserts or throws if it is already true), and a destructor and sets it to false like this

```
class TrampGuard  {
    public:
        TrampGuard(bool &b): guard(b) {assert(!guard); guard = true;}
        ~TrampGuard() {guard = false;}
    private
        bool &guard;
};
```

Then when you activate a guard then just create a local `TrampGuard` variable `TrampGuard trampGuard(inTramp);` instead of setting `inTramp` directly.  This will set `inTramp` and unset it when the scope exits.